Soleras3/react-challenge-project-jan-2023
# Feature: Log Out

## Changes
- Connect Log Out button to the logout auth action.
- Log Out routes user to the welcome (`Main`) screen.

## Purpose
Redux store should be cleared on log out. Sets up future features like route guarding.
See: https://github.com/Shift3/react-challenge-project-jan-2023/issues/5

## Approach
Action is handled in the Nav component where the Log Out button is.

## Testing Steps
1. Login
2. Click Log Out and observe the Welcome screen.

Testing Redux: After login, on Order Form, I changed the Log Out route to `#` so it stayed on screen. Then created an order and observed the user was "Unknown!" Would be neat to cover this with a test instead.

## Learning
I wasn't sure if I needed to navigate the user _after_ dispatching the logout action, so I looked that up. If it's better, I could have imported react router's `useHistory`. (I tried `useNavigate` first but that's a [v6 feature](https://reactrouter.com/en/main/upgrading/v5#use-usenavigate-instead-of-usehistory) apparently).
Closes https://github.com/Shift3/react-challenge-project-jan-2023/issues/5
